docs/dscom.build/ReadMe.md

```diff
@@ -0,0 +1,229 @@
+* `TargetFile` (Required): The full path to the type library to export.
+* `SourceAssemblyFile` (Required): The full path of the managed assembly to export.
+* `TypeLibraryReferences`: The input type libraries to reference during export.
+* `TypeLibraryReferencePaths`: A list of paths containing type library files to reference automatically during export.
+* `AssemblyPaths`: The references of the source assembly which must be loaded for export.
+
+It contains the method `bool Execute()` as an implementation of the `ITask` interface.
+
+The execute method will load the `dSPACE.Runtime.InteropServices.dll`. It tries to locate the file next to the build task assembly and - if cannot be present at that time - let the .NET runtime try to resolve it.
+
+During the execution phase, the following steps will be made:
+
+* Parse the overridden type library Guid.
+* Fail execution, if the build is not executed on windows.
+* Convert the `ITaskItems` (cf. `TypeLibraryReferences`, `TypeLibraryReferencePaths`, `AssemblyPaths`) to file system identifiers. If a file or directory is not present, a warning will be issued using the MsBuild `TaskLoggingHelper`. The export will continue.
+* Resolve the `SourceAssemblyFile` and issue an error, if it was not found.
+* Convert the assembly using the `DefaultBuildContext`.
+
+The `DefaultBuildContext` represents a level of abstraction. It contains the part of the implementation that cannot be stubbed or mocked at run-time.
+
+The execution will be successful, if the build context returns success and the task logging helper did not log any errors.
+
+### Class `DefaultBuildContext`
+
+The `DefaultBuildContext` is an implementation of the [`IBuildContext`](../../src/dscom.build/IBuildContext.cs) interface. It provides the following operations:
+
+* Check, whether the build is running on windows.
+* Check, whether given files and directories exist.
+* Perform the export using the `TypeLibConverterSettings`.
+
+The export is done using the `TypeLibConverter` instance and an implementation of the [`ITypeLibExporterNotifySink`](../../src/dscom.build/LoggingTypeLibExporterSink.cs), which logs the events to the build log.
+
+If the type library converter did not report a successful export, an error will be issued. If the expected file did not exists after the export, a warning shall be issued.
+
+It is an issue for discussion, if this check can be added to the `TlbExport` implementation. This way, the warning can be unit tested.
+
+### Build sequence (TlbExport)
+
+```ditaa
+                    ┌───────────┐                       ┌─────────┐                                                                     ┌─────────────┐                                                                                                         ┌─────────────────┐          ┌────────┐                                                                            ┌───┐                    

```

Comment:
Is the diagram something generated from the puml and txt file?

Reply:
The diagram is exactly the same as in the txt file.
I could reference it, if it were an asciidoc file.
The TXT file itself is generated from the puml file,